### PR TITLE
#VP-5555 Notification is not saved with a custom validation error

### PR DESCRIPTION
**Pull request by kostyrin**
https://virtocommerce.atlassian.net/browse/VP-5555

### Proposed of changes
Describe the technical details of realization provided by you.

### Make sure these boxes are checked:
- [ ] Check all the changes in github PR - files count (non of them are redundant, have meaningful changes, all are added), if target branch is correct
- [ ] Check methods and variable namings - it should be self descriptive, no typos
- [ ] Check you did not introduce breaking changes in API and public models/services.
- [ ] Respect extensibility - https://community.virtocommerce.com/t/extensibility-basics-the-domain-model-and-persistence-layer-extension/141
- [ ] Follow [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself) and [SOLID](https://en.wikipedia.org/wiki/SOLID) principles
- [ ] For unit tests - follow Microsoft best practices: https://docs.microsoft.com/en-us/dotnet/core/testing/unit-testing-best-practices
- [ ] Consolidate solution dependencies in case you are using newer version, update VC module dependencies in module.manifest. Do not upgrade 3rd party packages that are shipped with the platform with the version newer than in the platform.
- [ ] Check code style conventions - https://github.com/VirtoCommerce/styleguide/blob/master/csharp.md
- [ ] Check PR have a concise and descriptive title, follow [git commit message rules](https://github.com/VirtoCommerce/styleguide/blob/master/gitcommits.md)

### CODE REVIEW

**Review by akak1977 — APPROVED** [review]

**Review by tatarincev — COMMENTED** [review]
Why instead of all these changes shouldn't simply preserve the user error `CustomValidationError` for notification message during validation? It seems what the guys requested in the ticket.

**Comment by artem-dudarev** [comment]
`ScheduleSendNotificationAsync()` should not throw exceptions if `SetCustomValidationError()` was called. We already know that the notification is invalid and we just want to save it to the feed.

**Comment by kostyrin** [comment]
1. need to check Notification and Message before sending/scheduling
2. according to should not throw exceptions if there is CustomValidationError, then don't validate this kind of messages
3. added NotificationMessageExtensions with method 'CanSend'

File: src/VirtoCommerce.NotificationsModule.Data/Senders/NotificationSender.cs [hunk 1]
```diff
@@ -30,60 +31,35 @@ public class NotificationSender : INotificationSender
...
-
-                return await TrySendNotificationMessageAsync(message.Id);
-            }
-
-            return new NotificationSendResult();
+            return CheckAndTryToSendNotificationAsync(notification);
         }
 
-
+        //Must be a public method
```
Comment (tatarincev) [inline src/VirtoCommerce.NotificationsModule.Data/Senders/NotificationSender.cs:50]: TrySendNotificationMessageAsync(string messageId) is declared as public, that's mean now all previous checks are not executed here?

**Review by tatarincev — COMMENTED** [review]
I'm working on the own implementation of this feature because the current code is required rethinking and refactoring.

**Comment by kostyrin** [comment]
@tatarincev as you wish

**Pull request closed without merging**